x/bank/keeper/keeper.go

```diff
@@ -197,7 +203,13 @@ func (k BaseKeeper) UndelegateCoins(ctx sdk.Context, moduleAccAddr, delegatorAdd
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, amt.String())
 	}
 
-	err := k.subUnlockedCoins(ctx, moduleAccAddr, amt)
+	// call the BeforeSend hooks
+	err := k.BeforeSend(ctx, moduleAccAddr, delegatorAddr, amt)
+	if err != nil {
+		return err
```

Comment:
Should we really return an error if the hook fails? It could mean CW20 addresses may be non-transferable.

Reply:
With how we're using it in Osmosis, this is the goal.

For example, a tokenfactory token might want to enforce a blacklist.